shm/src/Store.cpp

```diff
@@ -44,27 +44,38 @@ Store::Response Store::buy(Cargo* cargo, size_t amount, Player* player) {
     if (player->getAvailableSpace() < amount) {
         return Response::lack_of_space;
     }
+    player->buy(cargo, amount, cargo->getPrice());
```

Comment:
no need to pass cargo->getPrice() as it is already easily accessible by cargo pointer (thus it is doubled here)